Have access to Fabric8 OpenShift Client

In my operator, if I detect that I'm running on OpenShift, I have specific resources to install (e.g. a `Route`).

I want to be able to use fabric8 OpenShiftClient in those cases (after detecting that I'm running on OpenShift) to do that with Java types to interact with these OpenShift resources .

I'm using the operator-framework-quarkus-extension to build my operator and if I inject a `io.fabric8.openshift.client.OpenShiftClient` it fails with

```
[ERROR]         [error]: Build step io.quarkus.arc.deployment.ArcProcessor#validate threw an exception: javax.enterprise.inject.spi.DeploymentException: javax.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type io.fabric8.openshift.client.OpenShiftClient and qualifiers [@Default]
[ERROR]         - java member: org.wildfly.operator.WildFlyOperator#client
[ERROR]         - declared on CLASS bean [types=[io.quarkus.runtime.QuarkusApplication, org.wildfly.operator.WildFlyOperator, java.lang.Object], qualifiers=[@Default, @Any], target=org.wildfly.operator.WildFlyOperator]
```

That'd be great if I could inject an OpenShift client (and maybe make it optional if the container platform is not detect to be OpenShift). Something like

```
    // always injected
    @Inject
    KubernetesClient client;

    // present only if the container platform is openshift
    @Inject
    Optional<OpenShiftClient> client;
```

--- a/operator-framework-quarkus-extension/deployment/pom.xml
+++ b/operator-framework-quarkus-extension/deployment/pom.xml
@@ -29,7 +29,7 @@
         </dependency>
         <dependency>
             <groupId>io.quarkus</groupId>
-            <artifactId>quarkus-kubernetes-client-deployment</artifactId>
+            <artifactId>quarkus-openshift-client-deployment</artifactId>
         </dependency>
         <dependency>
             <groupId>io.quarkus</groupId>

--- a/operator-framework-quarkus-extension/runtime/pom.xml
+++ b/operator-framework-quarkus-extension/runtime/pom.xml
@@ -29,7 +29,7 @@
     </dependency>
     <dependency>
       <groupId>io.quarkus</groupId>
-      <artifactId>quarkus-kubernetes-client</artifactId>
+      <artifactId>quarkus-openshift-client</artifactId>
     </dependency>
     <dependency>
       <groupId>io.quarkus</groupId>

--- a/operator-framework-quarkus-extension/tests/pom.xml
+++ b/operator-framework-quarkus-extension/tests/pom.xml
@@ -35,7 +35,7 @@
     </dependency>
     <dependency>
       <groupId>io.quarkus</groupId>
-      <artifactId>quarkus-test-kubernetes-client</artifactId>
+      <artifactId>quarkus-test-openshift-client</artifactId>
       <scope>test</scope>
     </dependency>
     <dependency>

--- a/samples/quarkus/src/main/java/io/javaoperatorsdk/operator/sample/QuarkusOperator.java
+++ b/samples/quarkus/src/main/java/io/javaoperatorsdk/operator/sample/QuarkusOperator.java
@@ -1,5 +1,6 @@
 package io.javaoperatorsdk.operator.sample;
 
+import io.fabric8.openshift.client.OpenShiftClient;
 import io.javaoperatorsdk.operator.Operator;
 import io.quarkus.runtime.Quarkus;
 import io.quarkus.runtime.QuarkusApplication;
@@ -10,6 +11,7 @@ import javax.inject.Inject;
 public class QuarkusOperator implements QuarkusApplication {
 
   @Inject Operator operator;
+  @Inject OpenShiftClient client;
 
   public static void main(String... args) {
     Quarkus.run(QuarkusOperator.class, args);

